File: backend/Models/Common/LLMConfig.cs
using System.ComponentModel.DataAnnotations;
using backend.Validation;

namespace backend.Models.Common
{
    public class LLMConfig
    {
        [Required]
        public required LLMProvider Provider { get; set; }
        [Required]
        [ValidModelForProvider]
        public required string Model { get; set; }
        [Required]
        public double Temperature { get; set; } = 0.7;
        [Required]
        public int MaxTokens { get; set; } = 200;
        public string? SystemPrompt { get; set;}
        public double? TopP { get; set; }
        public int? TopK { get; set; }
        public double? RepeatPenalty { get; set; }
        public int? Seed { get; set; }
        public bool Stream { get; set; } = true;
        public string[]? StopSequences { get; set; }
    }
}

File: frontend/src/domain/LLMConfig.ts
import type { LLMProvider } from "./enums/LLMProvider";

export type LLMConfig = {
    provider: LLMProvider;
    model: string;
    temperature: number;
    maxTokens: number;
    stream: boolean;
    systemPrompt?: string;
    topP?: number;
    topK?: number;
    repeatPenalty?: number;
    seed?: number;
    stopSequences?: string[];
};